### ISSUE TITLE
robotics: quarantine online controller adaptation behind qualification-preserving promotion

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#1430]
## Parent / program

Child of #1288. Integrates with #1398 qualification stages, #1413 applicability, #1414 change-impact/delta requalification, #1424 field-counterexample restriction, and task-specific operational admission such as Grasp #1401.

Also reuse the CogSec learning-promotion principle from #162/#210 where applicable: learning artifacts may be observed, trained, aggregated and qualified, but must not grant themselves authority to replace a trusted production controller.

## Concrete current seam

`symthaea-humanoid` already contains a mutable training subsystem with controller checkpoints/resume bundles and adaptive curriculum. The broader cognitive runtime also has optional online CfC learning during inference.

Those are useful research capabilities. Before any such mechanism can influence a live embodied controller, freeze a separate production theorem:

> adaptation produces a new candidate identity; it does not mutate the currently qualified operational candidate in place.

## Core invariant

```text
controller C0 earned qualification Q0
        +
online/offline update Δ
        -> candidate C1

C1 != C0 for qualification/authority purposes
```

unless an explicit prequalified adaptation theorem proves that the exact state transition remains within a bounded qualified adaptation envelope.

Default high-assurance rule:

```text
controller mutation
    -> old operational admission invalid for mutated controller
    -> new candidate is quarantined
    -> #1414 determines required Simulation/HIL/Physical delta/full evidence
    -> only a newly admissible qualified candidate may later receive live authority
```

No successful runtime reward, prediction improvement, low loss, high Φ/Ψ, operator convenience, or absence of immediate failures may substitute for this transition.

## Separate three kinds of adaptation

Do not put all changing state under one `adaptive=true` flag.

### 1. Epistemic / plant-state adaptation

Examples:

- state estimator posterior updates;
- body-schema estimates;
- payload estimate;
- terrain estimate;
- calibration-health estimate;
- actuator health/efficiency estimate;
- learned but non-authoritative world-model state.

These update **belief/evidence state**. They may restrict action and may feed explicit safety/admission policies, but they are not automatically controller-policy mutation.

### 2. Ephemeral bounded control state

Examples:

- integrator/recurrent hidden state;
- contact state;
- warm-start state;
- adaptive observer state;
- explicitly qualified gain-scheduling state.

This state may evolve during an action only when its exact semantics and bounds are part of the qualified controller definition. Reset/restore/recurrent-state semantics must remain explicit.

### 3. Persistent policy/controller mutation

Examples:

- learned weights;
- LoRA/delta weights;
- policy-head weights;
- persistent gains;
- optimizer-applied parameters;
- learned action mapping;
- persistent time constants when behaviorally material;
- generated/replaced controller code/configuration.

These create a new qualification candidate by default.

Do not allow a persistent mutation to masquerade as ordinary hidden/control state.

## Immutable active candidate

Operational admission must bind one exact controller/candidate identity for the complete admission/lease interval.

Once an action is admitted for candidate `C0`:

- the controller artifact/configuration used for goal-directed execution may not change to `C1` under that admission;
- any persistent adaptation result becomes an inert candidate artifact;
- candidate swap requires a new qualified/current controller binding and a new operational admission;
- continuation-lease renewal must confirm the active candidate identity remains unchanged.

A candidate change before effect-entry linearization invalidates the admission. A discovered mutation after effect entry is an evidence-bearing discontinuity and must enter the qualified interruption/fallback policy, not continue silently.

## Adaptation candidate artifact

Introduce/reuse a canonical inert artifact conceptually named:

`EmbodiedControllerAdaptationCandidateV1`

committing, at minimum:

- schema/version;
- exact parent controller/candidate digest;
- exact parent qualification-subject snapshot from #1414;
- exact adaptation mechanism/algorithm identity;
- exact optimizer/update-rule identity;
- exact training/adaptation configuration;
- exact input/evidence/training-corpus lineage or canonical root;
- exact randomness/seed profile where material;
- exact starting mutable parameter-state digest;
- exact resulting controller parameter/artifact digest;
- exact changed semantic component set;
- creation execution/toolchain environment lineage;
- producer/provenance identity;
- candidate digest.

The artifact is ordinary quarantined data. It has no motor authority and no public conversion to a qualified controller.

## Training-data / experience lineage

A robot must not be able to hide what experience caused a physical controller change.

For consequential adaptation, retain exact provenance for the observations/episodes/rewards/errors that contributed to the candidate, including enough dependence identity to avoid counting replayed/duplicated experience as independent evidence.

Remote/federated/peer-supplied learning material additionally inherits CogSec #162/#210 constraints:

- sender self-reported trust is non-authoritative;
- remote updates remain quarantined;
- authentication != correctness != promotion authority;
- no remote gradient/LoRA/delta may directly alter the active embodied controller.

## Reward/optimization is not safety policy

Training reward and loss functions are optimization objectives, not authority or physical-safety proofs.

Explicitly freeze:

```text
higher reward != safer
lower prediction error != authorized
successful episode != qualification
more samples != independent assurance
```

If safety constraints are included in training, they may shape the candidate, but live positive safety/authority still comes from the qualified safety/admission stack.

## Default candidate promotion path

For an adapted candidate `C1`, route through #1414:

```text
C0 qualification subject snapshot
+ C1 subject snapshot
+ exact semantic change set
+ claim dependency profile
-> impact assessment
-> required delta/full campaign plan
-> new stage-specific evidence
-> new qualification artifact/envelope
-> current #1424 disposition
-> later live operational admission
```

A Simulation-only requalification cannot refresh HIL/Physical qualification. Each affected stage follows #1398 semantics.

## Bounded prequalified adaptation envelope

Some adaptive control techniques legitimately change parameters online. Do not ban them categorically.

Instead require an explicit optional theorem conceptually named:

`QualifiedControllerAdaptationEnvelopeV1`.

It may permit a narrowly defined runtime parameter evolution without treating each control tick as a brand-new unqualified controller only if the complete **adaptation law itself** was part of the qualified candidate.

The envelope must precommit, as applicable:

- exact mutable parameter/state set;
- exact immutable parameter/state set;
- exact adaptation/update law + implementation digest;
- parameter bounds and units;
- rate-of-change/update-frequency bounds;
- projection/clamping semantics;
- triggering observations/evidence classes;
- reset/initialization semantics;
- persistence semantics across reboot/action boundaries;
- applicability domain in which the adaptation law was qualified;
- exact safety monitor/invariant obligations;
- exact fallback/freeze behavior when bounds/evidence are violated;
- exact Simulation/HIL/Physical campaigns that exercised the adaptation law across the permitted envelope;
- adaptation-envelope digest.

The positive qualification claim is then about **the bounded adaptive controller as a state machine**, not about one frozen parameter vector.

Do not create a generic `adaptive=true` exemption.

## No self-widening adaptation

The adaptation mechanism must never be able to widen its own:

- parameter bounds;
- applicability scope;
- safety limits;
- qualification stage;
- authority cap;
- update-rate limits;
- trusted input classes;
- promotion policy.

Those are external immutable policy/qualification commitments.

An update trying to leave the qualified envelope must be projected/rejected/frozen according to the exact qualified policy and cannot continue goal-directed operation under the old claim.

## Runtime adaptation receipt

For a prequalified bounded adaptive controller, each material adaptation transition should emit an evidence object conceptually named:

`ControllerAdaptationTransitionReceiptV1`

binding:

- exact adaptive-controller/envelope identity;
- exact prior adaptive-state/parameter digest;
- exact triggering evidence digest(s);
- exact update-law identity;
- exact proposed parameter delta/state transition;
- exact accepted/projected resulting state;
- exact bound/invariant checks;
- exact resulting parameter/state digest;
- transition sequence/generation;
- timestamp under explicit clock semantics where needed;
- receipt digest.

A missing receipt for a required persistent/material update is a qualification/operational discontinuity, not a harmless logging omission.

## Learning and safety monitor separation

A learning mechanism must not be the sole verifier of the safety invariant it is trying to optimize.

Where the adaptation envelope requires safety bounds, use deterministic/independently qualified runtime enforcement outside the mutable learner where practical.

The learner may propose a parameter update; the adaptation guard validates/projects/rejects it according to immutable qualified policy.

This parallels the existing principle that cognition may request/propose action but cannot mint its own plant-safety authority.

## Shadow-first adaptation

For persistent candidate updates not covered by a prequalified runtime adaptation envelope, prefer:

```text
active C0 remains unchanged
    +
candidate C1 runs in shadow/simulation/replay
    -> evidence
    -> #1414 impact/requalification
    -> explicit protected promotion
```

Shadow evaluation must not share physical effect authority with the candidate under test.

If candidate C1 observes live plant data, observation access is not actuation authority.

## Exploration boundary

Online learning often requires exploration. Exploration is physical action and therefore requires ordinary action authority/safety; a training objective cannot authorize exploratory motion by itself.

In production-shaped operation:

- exploration actions must lie inside current Physical qualification/applicability and live plant-safety/mission authority;
- elevated-risk exploration requires an explicit test/qualification mission profile;
- the learner cannot deliberately violate the safety envelope to gather informative data;
- qualification campaigns may use controlled exploration inside their isolated Simulation/HIL/Physical test scopes under separate qualification authority.

## Body-schema / system-identification updates

Keep learning about the robot separate from changing how the robot acts.

A new body-schema/plant estimate may become current evidence through its own estimator/calibration validation policy. If that new estimate implies the current controller is outside its qualified assumptions—for example changed payload, friction, actuator strength or kinematic calibration—then live authority must restrict and #1424/#1414 may be triggered.

Do not automatically retune the controller and continue under the old Physical qualification.

## Drift and counterexamples

If bounded adaptation repeatedly approaches limits, causes rising fallback/intervention rates, violates expected performance, or produces a credible field counterexample:

- preserve exact adaptation-transition lineage;
- allow #1424 to narrow/suspend current qualification;
- freeze goal-directed adaptation/control as policy requires;
- retain qualified local protective fallback;
- hand off to #1414 for change-impact/requalification.

Successful future adaptations cannot vote away an admitted counterexample.

## Rollback semantics

Rolling back from C1 to previously qualified C0 does not automatically restore operational admission.

Current physical state/calibration/wear/context may have changed while C1 was active. Re-establish the exact current qualification disposition, active controller identity, provider/health/calibration state, plant safety and one-use operational authority before new goal-directed action.

Historical C0 qualification remains evidence, not a reusable live permit.

## Persistence / reboot

Persistent adaptive state must have explicit checkpoint identity and restore semantics.

A reboot must not silently:

- load unqualified learned parameters;
- reset an adaptive controller to optimistic/default bounds;
- erase evidence that the controller changed;
- restore prior operational authority;
- merge adaptive states from incompatible parent candidates.

If the qualified adaptation law allows persistent state, checkpoint/restore identity is part of the qualified state machine. Otherwise restart at the exact qualified immutable candidate and treat learned state as quarantined data pending promotion.

## Regression requirements

At minimum:

1. mutating one persistent controller parameter changes candidate identity by default;
2. an adapted candidate cannot reuse the parent's Physical envelope as if unchanged;
3. active operational admission binds immutable candidate identity;
4. candidate mutation before effect-entry invalidates admission;
5. candidate mutation during a sustained action blocks later continuation lease and enters qualified interruption/fallback semantics;
6. training reward improvement cannot mint qualification or motor authority;
7. online learning flag alone cannot exempt controller changes from qualification;
8. body-schema/estimator update can restrict operation without mutating the controller;
9. uncontrolled auto-retuning after plant/body estimate change is forbidden in the strongest production path;
10. quarantined C1 may run in shadow without gaining physical effect authority;
11. remote/federated update cannot directly mutate active controller;
12. sender-provided trust cannot increase update promotion privilege;
13. exact #1414 change set is derived from C0/C1 snapshots rather than caller-declared `small_change`;
14. Simulation delta evidence cannot refresh affected Physical qualification;
15. a prequalified adaptation envelope accepts only exact declared mutable parameters/update law/bounds;
16. update outside adaptation bounds is rejected/frozen and cannot self-widen the envelope;
17. adaptation transition receipt binds prior -> resulting parameter/state identity and triggering evidence;
18. duplicate/replayed update cannot create multiple independent evidence or repeated persistent mutation without explicit semantics;
19. missing required adaptation receipt causes current qualification/admission failure where policy requires it;
20. credible adaptation-related counterexample can trigger #1424 restriction while preserving historical qualification;
21. rollback to old weights does not restore old live admission automatically;
22. reboot cannot load learned state from another parent candidate/envelope;
23. learner cannot authorize exploratory motion outside ordinary mission/safety/qualification scope;
24. local protective fallback remains available when goal-directed adapted control is frozen;
25. no adaptation artifact/receipt exposes motor lowering or creates Operational authority.

## Recommended initial implementation

Do not start with live motor adaptation.

After #1398/#1413/#1414 have executable signal:

1. canonical immutable active-controller identity + adaptation candidate artifact;
2. static/runtime guard proving active operational candidate cannot mutate in place;
3. shadow candidate evaluation + #1414 handoff;
4. checkpoint/restore anti-substitution tests;
5. only then prototype one tiny prequalified adaptive-state envelope in Simulation;
6. require dedicated HIL/Physical campaigns before allowing that adaptive law in corresponding stages.

A useful first adaptive-law experiment would be a tightly bounded, deterministic parameter such as a qualified observer/gain schedule with explicit projection bounds—not unconstrained neural weight learning.

## CI / status

Keep this specification-first while repository Actions remain heavily queued. Do not create a live adaptation implementation branch merely to exercise the idea without exact-head signal.

## Safety / non-claims

This issue does not claim current humanoid training is unsafe or production-connected. It defines the production boundary needed before learning/adaptation can control physical hardware. It does not ban adaptive control, prove a specific adaptation law safe, authorize exploratory physical behavior, or grant motor authority.
